pkg/querier/queryrange/labels_cache.go

```diff
@@ -11,21 +11,42 @@ import (
 	"github.com/grafana/loki/pkg/querier/queryrange/queryrangebase"
 	"github.com/grafana/loki/pkg/storage/chunk/cache"
 	"github.com/grafana/loki/pkg/storage/chunk/cache/resultscache"
-	"github.com/grafana/loki/pkg/util"
+	"github.com/grafana/loki/pkg/util/validation"
 )
 
 type cacheKeyLabels struct {
 	Limits
 	transformer UserIDTransformer
-	iqo         util.IngesterQueryOptions
+}
+
+// metadataSplitIntervalForTimeRange returns split interval for series and label requests.
+// If `recent_metadata_query_window` is configured and the query start interval is within this window,
+// it returns `split_recent_metadata_queries_by_interval`.
+// For other cases, the default split interval of `split_metadata_queries_by_interval` will be used.
```

Comment:
I actually think your code is pretty simple and self-documenting; all this comment can do is go stale.